src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java

```diff
@@ -84,6 +85,15 @@ public MetricsDB(long windowStartTime) throws Exception {
     create = DSL.using(conn, SQLDialect.SQLITE);
   }
 
+  public static MetricsDB fetchExisting(long windowStartTime) throws Exception {
```

Comment:
We only marked those other callers as "VisibleForTesting" because we expanded their visibility from private to public simply for the sake of testing. However, even if they were private, this method would have to have public visibility for them to be able to access it.

Reply:
That might be misleading. That method should not be called from anywhere in the code other than tests. For your usecase, you can make that protected and then have a wrapper Test class that exposes it over a public interface for tests only ?

Reply:
Will address this by making the target methods private and exposing them via VisibleForTesting shims.